lib/ecl/smspec_node.cpp

```diff
@@ -439,73 +441,147 @@ char * smspec_alloc_local_completion_key( const char * join_string, const std::s
 }
 
 /*****************************************************************/
+static bool match_keyword_vector(int ipos, const std::vector<std::string> &vars, const std::string &keyword) {
```

Comment:
Type of `ipos` should be `size_t`, rather than `int`. C++ standard classes (ie. `std::string`, `std::vector`) use `size_t` (unsigned 64-bit ints) to denote size. It doesn't make sense to ask for a negative position, so using `int` (signed 32-bit int) is incorrect.

Note that this is a matter of correctness and will most likely generate the same machine code. The compiler should've warned you that you're comparing an unsigned to a signed value when compiling this.